tests/com/twitter/intellij/pants/integration/OSSPantsCompileActionsTest.java

```diff
@@ -77,6 +86,36 @@ public void testCompileAllTargetsInModuleAction() throws Throwable {
     assertFalse(compileAllTargetsInModuleAction.doCleanAll());
   }
 
+  public void testCompileTargetsInSelectedEditor() throws Throwable {
+    doImport("examples/tests/scala/org/pantsbuild/example");
+    ArrayList<Pair<String, String>> testClassAndTarget = Lists.newArrayList(
+      // Pair of class reference and its containing target
+      Pair.create(
+        "org.pantsbuild.example.hello.welcome.WelSpec",
+        "examples/tests/scala/org/pantsbuild/example/hello/welcome:welcome"
+      ),
+      Pair.create(
+        "org.pantsbuild.example.hello.welcome.WelcomeEverybody",
+        "examples/src/scala/org/pantsbuild/example/hello/welcome:welcome"
+      )
+    );
+
+    for (Pair<String, String> classAndTarget : testClassAndTarget) {
+      String clazzName = classAndTarget.getFirst();
+      String target = classAndTarget.getSecond();
+
+      PsiClass clazz = JavaPsiFacade.getInstance(myProject)
+        .findClass(clazzName, GlobalSearchScope.projectScope(myProject));
+
+      assertNotNull(String.format("%s does not exist, but should.", clazz), clazz);
+
+      FileEditorManager.getInstance(myProject).openFile(clazz.getContainingFile().getVirtualFile(), true);
+      PantsCompileCurrentTargetAction compileCurrentTargetAction = new PantsCompileCurrentTargetAction();
+      Set<String> currentTargets = compileCurrentTargetAction.getTargets(getPantsActionEvent(), myProject).collect(Collectors.toSet());
```

Comment:
Could you adjust the whitespace here a bit to make it clearer that the `getTargets` call is what is exercising the code being tested?